pkg/domain/serverinfo/syncer.go

```diff
@@ -0,0 +1,442 @@
+}
+
+// UpdateServerLabel updates the labels of the local server information in etcd.
+func (s *Syncer) UpdateServerLabel(ctx context.Context, labels map[string]string) error {
+	// when etcdCli is nil, the server infos are generated from the latest config, no need to update.
+	if s.etcdCli == nil {
+		return nil
+	}
+	dynamicInfo := s.cloneDynamicServerInfo()
+	changed := false
+	for k, v := range labels {
+		if dynamicInfo.Labels[k] != v {
+			changed = true
+			dynamicInfo.Labels[k] = v
+		}
+	}
+	if !changed {
+		return nil
+	}
+	info := s.GetLocalServerInfo().Clone()
+	info.DynamicInfo = *dynamicInfo
+	infoBuf, err := info.Marshal()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	str := string(hack.String(infoBuf))
+	err = util.PutKVToEtcd(ctx, s.etcdCli, KeyOpDefaultRetryCnt, s.serverInfoPath, str, clientv3.WithLease(s.session.Lease()))
+	if err != nil {
+		return err
+	}
+	// update the dynamic info in the global info syncer after put etcd success.
+	s.setDynamicServerInfo(dynamicInfo)
+	return nil
+}
+
+// cloneDynamicServerInfo returns a clone of the dynamic server info.
+func (s *Syncer) cloneDynamicServerInfo() *DynamicInfo {
+	return s.info.Load().DynamicInfo.Clone()
+}
+
+// setDynamicServerInfo updates the dynamic server info.
+func (s *Syncer) setDynamicServerInfo(ds *DynamicInfo) {
+	staticInfo := s.info.Load()
+	newInfo := &ServerInfo{
+		StaticInfo:  staticInfo.StaticInfo,
+		DynamicInfo: *ds,
+	}
+	s.info.Store(newInfo)
+}
+
+// GetAllServerInfo returns all server information from etcd.
+func (s *Syncer) GetAllServerInfo(ctx context.Context) (map[string]*ServerInfo, error) {
+	allInfo := make(map[string]*ServerInfo)
+	if s.etcdCli == nil {
+		info := s.info.Load()
+		allInfo[info.ID] = getServerInfo(info.ID, info.ServerIDGetter)
+		return allInfo, nil
+	}
+	allInfo, err := getInfo(ctx, s.etcdCli, ServerInformationPath, KeyOpDefaultRetryCnt, KeyOpDefaultTimeout, clientv3.WithPrefix())
+	if err != nil {
+		return nil, err
+	}
+	return allInfo, nil
+}
+
+// Done returns a channel that closes when the info syncer is no longer being refreshed.
+func (s *Syncer) Done() <-chan struct{} {
+	if s.etcdCli == nil {
+		return make(chan struct{}, 1)
+	}
+	return s.session.Done()
+}
+
+// Restart the info syncer with new session leaseID and store server info to etcd again.
+func (s *Syncer) Restart(ctx context.Context) error {
+	return s.NewSessionAndStoreServerInfo(ctx)
+}
+
+// RemoveServerInfo remove self server static information from etcd.
+func (s *Syncer) RemoveServerInfo() {
+	if s.etcdCli == nil {
+		return
+	}
+	err := util.DeleteKeyFromEtcd(s.serverInfoPath, s.etcdCli, KeyOpDefaultRetryCnt, KeyOpDefaultTimeout)
+	if err != nil {
+		logutil.BgLogger().Error("remove server info failed", zap.Error(err))
+	}
+}
+
+// ServerInfoSyncLoop syncs the server information periodically.
+func (s *Syncer) ServerInfoSyncLoop(store tidbkv.Storage, exitCh chan struct{}) {
+	defer func() {
+		logutil.BgLogger().Info("infoSyncerKeeper exited.")
```

Comment:
Maybe update the log part in this file?
```suggestion
		logutil.BgLogger().Info("ServerInfoSyncer exited")
```